include/highfive/bits/H5Reference_misc.hpp

```diff
@@ -43,7 +43,7 @@ inline T Reference::dereference(const Object& location) const {
         HDF5ErrMapper::ToException<ReferenceException>(
             "Trying to dereference the wrong type");
     }
-    return std::move(obj);
+    return obj;
```

Comment:
we should keep the move because the types are not the same. Without move a copy will happen

Reply:
gcc 10.2 fail to compile with the move (due to -Werror)

Reply:
but fixed